src/csharp/Grpc.Core/Internal/PlatformInvocation.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Grpc.Core.Internal
+{
+    /// <summary>
+    /// Platform invocation dispatch
+    /// </summary>
+    internal static class PlatformInvocation
+    {
+        /// <summary>
+        /// Retrieve an implementation of platform invocations specific to the current cpu's architecture.
+        /// This allows assemblies to function correctly when compiled for AnyCPU
+        /// </summary>
+        internal static IPlatformInvocation Implementation { get; private set; }
```

Comment:
The only difference is the binding for the DllImport annotation.  Without this, you can only use this library from a matching architecture and platform, which is 32-bit Windows.  After compiling with this change, I was able to compile my driver with AnyCPU and the same build worked in all use-cases (most notably Linux as that is my primary target, although I would like Bigtable.NET to just work in all cases).  Pragmas would have worked but would require different builds.  Also, with an interface the native functionality can be mocked for testing, so I opted for this approach.